### PR TITLE
Do not save questions in localstore

### PR DESCRIPTION
**Pull request by Aaron2000119**
Closes https://github.com/Gamify-IT/issues/issues/146.
Questions are stored in a vuex store.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -2,6 +2,7 @@ module.exports = {
   root: true,
   env: {
     node: true,
+    "vue/setup-compiler-macros": true,
```
Comment (maexled) [inline .eslintrc.js:5]: What does this?

File: src/store/index.ts [hunk 2]
```diff
@@ -1,9 +1,21 @@
...
-  modules: {},
+export interface State {
+  questions: Array<Question>;
+}
+
+export const key: InjectionKey<Store<State>> = Symbol();
+
+export const store = createStore<State>({
+  state: {
+    questions: questionsJson,
```
Comment (maexled) [inline src/store/index.ts:14]: You put a default? Is this really the best option? What is about error handling? I will mention this in an extra comment.

**Review by maexled — COMMENTED** [review]
You need to do a error handling if requested config does not exist and backend and do not let the user start the game trough the button.

File: .eslintrc.js (hunk 1 above)
Comment (Aaron2000119) [reply .eslintrc.js:5]: With the new setup script, you can use defineProps without defining it. With an old vue-eslint-parser, this generates an no-undef error wich is disabled with this line.

**Review by maexled — COMMENTED** [review]
Game is still startable when the error is displayed. It just leaves a blank screen on the Crosswordpuzzle view. There also needs to be an error handling

File: src/views/HomeView.vue [hunk 3]
```diff
@@ -1,35 +1,57 @@
...
 import config from "@/config";
 import { ref } from "vue";
+import { store } from "@/store/index";
+import questionsJson from "@/assets/questions.json";
 
 let questions = ref(Array<Question>());
 
+let errorText = ref("");
+
+let isActive = false;
```
Comment (maexled) [inline src/views/HomeView.vue:14]:
```suggestion
const isActive = ref(false);
```
needs to be a ref, because state is changing during rendering

**Review by maexled — COMMENTED** [review]
Currently the questions does not show because isActive is not a ref

File: src/views/HomeView.vue [hunk 4]
```diff
@@ -1,35 +1,57 @@
...
+      isActive.value = true;
     })
-    .then(() => {
-      localStorage.setItem("questions", JSON.stringify(questions.value));
-      console.log(questions.value);
+    .catch((error) => {
+      if (error.response) {
+        console.log(error.response.data);
+        console.log(error.response.status);
+        console.log(error.response.headers);
```
Comment (delvh) [inline src/views/HomeView.vue:36]:
```suggestion
        console.log(error.response.headers);
        isActive = false;
```

**Review by delvh — APPROVED** [review]

**Pull request merged**